src/request/swap-iframe/SwapIFrameApi.js

```diff
@@ -0,0 +1,385 @@
+/* global BitcoinRequestParserMixin */
+/* global RequestParser */
+/* global Nimiq */
+/* global loadNimiq */
+/* global BitcoinJS */
+/* global BitcoinUtils */
+/* global HtlcUtils */
+/* global Errors */
+
+class SwapIFrameApi extends BitcoinRequestParserMixin(RequestParser) {
+    /**
+     * @param {RpcState?} state
+     * @param {KeyguardRequest.SignSwapTransactionsRequest} request
+     * @returns {Promise<KeyguardRequest.SignSwapTransactionsResult>}
+     */
+    async signSwapTransactions(state, request) {
+        const storageKey = SwapIFrameApi.SESSION_STORAGE_KEY_PREFIX + request.swapId;
+
+        const storedData = sessionStorage.getItem(storageKey);
+        sessionStorage.removeItem(storageKey); // Delete storage
+
+        if (!storedData) throw new Error('No swap stored in SessionStorage');
+
+        /** @type {{keys: {nim: string, btc: string[]}, request: any}} */
+        const { keys: privateKeys, request: storedRawRequest } = (JSON.parse(storedData));
+
+        if (request.fund.type === 'NIM' || request.redeem.type === 'NIM') {
+            if (!privateKeys.nim) throw new Error('No NIM key stored in SessionStorage');
+            if (privateKeys.nim.length !== 64) throw new Error('Invalid NIM key stored in SessionStorage');
+        }
+
+        if (request.fund.type === 'BTC' || request.redeem.type === 'BTC') {
+            if (!privateKeys.btc) throw new Error('No BTC key list stored in SessionStorage');
+            if (!privateKeys.btc.length) throw new Error('No BTC keys stored in SessionStorage');
+            if (privateKeys.btc.some(key => !key)) throw new Error('Empty BTC key stored in SessionStorage');
+            if (privateKeys.btc.some(key => key.length !== 64)) {
+                throw new Error('Invalid BTC key stored in SessionStorage');
+            }
+        }
+
+        // Deserialize stored request
+        /** @type {Parsed<KeyguardRequest.SignSwapRequest>} */
+        const storedRequest = {
```

Comment:
Again this could be done with less inlining for a more readable code in my opinion.